tox.ini

```diff
@@ -1,32 +1,22 @@
 [tox]
-envlist = py27,py34,py35,py36,pypy,pypy3,coverage,docs
+envlist =
+   py27,py34,py35,py36,pypy,pypy3,coverage,docs
 
 [testenv]
-# We need to use develop egg to get namespace directories right.
-usedevelop = true
 commands =
-    python setup.py -q test -q
+    zope-testrunner --test-path=src []
+    sphinx-build -b doctest -d {envdir}/doctrees docs {envdir}/doctest
 deps =
-    python-gettext
-    pytz
-    zope.component
-    zope.configuration
-    zope.i18nmessageid
-    zope.location
-    zope.proxy
-    zope.schema
-    zope.security
-    zope.testing
-    zope.testrunner
+    .[test,docs,compile]
```

Comment:
I recently learned that `tox` understands `extras = test,docs,compile`.

Reply:
Interesting, I did not know that.

At first glance, I think I still prefer the `deps=.[foo,bar]` syntax. It's what RTD (and requirements files in general) understands, so it's consistent with that. It also seems more explicit to me to have that leading `.`.

Reply:
And I just learned that tox will allow `deps=-rrtd.txt` so in theory we could DRY there too (though I'm not sure that's necessarily a good idea in general).